Helper/Entity/Product/PriceManager.php

```diff
@@ -503,26 +507,108 @@ private function handleZeroDefaultPrice(
         $baseCurrencyCode,
         $min,
         $max,
-        Store $store
+        Store $store,
+        $subProducts,
+        $areCustomersGroupsEnabled,
+        $groups,
+        $specialPrice
     ) {
+        if ($min == $max) {
+            if (!$areCustomersGroupsEnabled && !empty($specialPrice)) {
+                $min = $max = $specialPrice[0];
+            } elseif ($areCustomersGroupsEnabled && !empty($specialPrice)) {
+                foreach ($groups as $group) {
+                    $groupId = (int)$group->getData('customer_group_id');
+
+                    $customData[$field][$currencyCode]['group_' . $groupId] = $specialPrice[$groupId];
+                    $groupSpecialPriceFormated = $this->priceCurrency->format(
+                        $specialPrice[$groupId],
+                        false,
+                        PriceCurrencyInterface::DEFAULT_PRECISION,
+                        $store,
+                        $currencyCode
+                    );
+                    $customData[$field][$currencyCode]['group_' . $groupId . '_formated'] = $groupSpecialPriceFormated;
+                }
+            }
+        }
+
         $customData[$field][$currencyCode]['default'] = $min;
 
+        $defaultOriginals = [];
+        foreach ($subProducts as $subProduct) {
+            $defaultOriginals[] = $subProduct->getPrice();
+        }
+
+        $defaultOriginal = min($defaultOriginals);
```

Comment:
This fails when there are no `$subProducts`. And then it has consequences to further code.

Reply:
Right, that makes no sense. Maybe we should add and `if empty($subProducts) { $defaultOriginals[] = <original product price here>; }` ?